Index: modules/simple_ldap_user/simple_ldap_user.module
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/modules/simple_ldap_user/simple_ldap_user.module b/modules/simple_ldap_user/simple_ldap_user.module
--- a/modules/simple_ldap_user/simple_ldap_user.module	(revision 20d247438b618073701e2c38f2867c3d72d06923)
+++ b/modules/simple_ldap_user/simple_ldap_user.module	(date 1614167439067)
@@ -30,9 +30,13 @@
   /** @var \Drupal\user\UserInterface $user */
   $user = $form_state->getFormObject()->getEntity();
 
+  // Check whether the user exists on LDAP.
+  $manager = \Drupal::service('simple_ldap_user.manager');
+  $ldap_user = $manager->getLdapUser($user->getAccountName());
+
   // Lock down the fields if applicable
   $readonly = FALSE;
-  if ($server->isReadOnly() && !$user->isNew() && $user->id() != 1) {
+  if ($server->isReadOnly() && !empty($ldap_user) && !$user->isNew() && $user->id() != 1) {
     \Drupal::messenger()->addWarning(
       t('Some fields have been marked <em>read-only</em>, as they are controlled by the LDAP server.')
     );
